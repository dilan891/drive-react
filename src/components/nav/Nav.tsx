import 'bootstrap/dist/css/bootstrap.min.css';
import "../../css/nav.css";
import { Link } from "react-router-dom";
import SelectVar from "./SelectVar"
import React from 'react';

const Nav: React.FC = () => {  //barra de navegacion de la pagina 

    const singOut = () =>{
        localStorage.removeItem("jwtToken");
    }

    return (
        <nav>
            <div className="user"><h1>Dilan891</h1></div>
            <button type="button" className="sing out" onClick={singOut}>cerrar seccion</button>
            <div className="navegation">
                <ul className="nav nav-tabs">
                    <li className="nav-item">
                        <Link to="/menu/Todos"
                            id="todo"
                            className="nav-link"
                            style={{ color: "black" }}
                            onClick={() => tab("todo")}>
                            Todos</Link>
                    </li>
                    <li className="nav-item">
                        <Link
                            to="/menu/carpetas"
                            id="prueba"
                            className="nav-link"
                            style={{ color: "black" }}
                            onClick={() => tab("prueba")}>
                            Carpetas</Link>
                    </li>
                </ul>
                <SelectVar />
            </div>
        </nav>
    );
}

function tab(name: string) {
    let tablinks
    tablinks = document.getElementsByClassName("nav-link");
    for (let i = 0; i < tablinks.length; i++) {
        tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(name)!.classList.add("active");
}

export default Nav;
