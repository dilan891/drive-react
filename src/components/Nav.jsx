import 'bootstrap/dist/css/bootstrap.min.css';
import "../css/nav.css";
import {Link} from "react-router-dom";
import SelectVar from "./SelectVar"

function Nav(){  //barra de navegacion de la pagina 
    return (
        <nav>
            <div className="user"><h1>Dilan891</h1></div>
            <div className="navegation">
              <ul className="nav nav-tabs">
                  <li className="nav-item">
                      <Link to="/Todos" 
                      id="todo" 
                      className="nav-link" 
                      style={{color:"black"}} 
                      onClick={()=> tab("todo")}>
                          Todos</Link>
                  </li>
                  <li className="nav-item">
                      <Link 
                      to="/carpetas" 
                      id="prueba"
                       className="nav-link" 
                        style={{color:"black"}}
                        onClick={()=> tab("prueba")}>
                            carpetas</Link>
                  </li>
              </ul>
              <SelectVar /> 
            </div>
        </nav>
    );
}

function tab(name){
    let tablinks
    tablinks = document.getElementsByClassName("nav-link");
    for (let i = 0; i < tablinks.length; i++) {
        tablinks[i].className = tablinks[i].className.replace(" active", "");
    }
    document.getElementById(name).classList.add("active");
}

export default Nav;
