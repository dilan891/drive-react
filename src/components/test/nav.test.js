import React from "react";
import Nav from "../Nav"
import '@testing-library/jest-dom'
import { BrowserRouter as Router } from 'react-router-dom';
import {screen,render} from "@testing-library/react"

beforeEach(()=>render(<Router><Nav /></Router> ))  //componente a renderizar
describe("NavPage", ()=> {//priemro el nombre de la prueba y luego las pruebas
    it("display todos page",()=>{
        expect(screen.queryByText(/Todos/i)).toBeInTheDocument()
    })
    it("display name user",()=>{
        expect(screen.queryByText(/Dilan891/i)).toBeInTheDocument()
    })
})  

