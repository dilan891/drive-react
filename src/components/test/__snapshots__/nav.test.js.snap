// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavPage snapshot nav 1`] = `
<div>
  <nav>
    <div
      className="user"
    >
      <h1>
        Dilan891
      </h1>
    </div>
    <div
      className="navegation"
    >
      <ul
        className="nav nav-tabs"
      >
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/Todos"
            id="todo"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
              }
            }
          >
            Todos
          </a>
        </li>
        <li
          className="nav-item"
        >
          <a
            className="nav-link"
            href="/carpetas"
            id="prueba"
            onClick={[Function]}
            style={
              Object {
                "color": "black",
              }
            }
          >
            carpetas
          </a>
        </li>
      </ul>
      <div
        className="select-var"
        style={
          Object {
            "position": "absolute",
            "visibility": "hidden",
          }
        }
      >
        <div>
          seleccionaste 
          <strong />
           
          moverlo a null
        </div>
        <button
          className="btn btn-info"
          id="b-dis"
          type="button"
        >
          Mover aqui
        </button>
        <button
          className="btn btn-info"
          onClick={[Function]}
          type="button"
        >
          X
        </button>
      </div>
    </div>
  </nav>
</div>
`;
